### PR TITLE
fix: #523, Select: display "label" in place of "value"

### PR DESCRIPTION
**Pull request by jacargentina**
First contribution, please be kind 😊, I can fix anything needed on this PR. Cheers!

### CODE REVIEW

**Comment by Viraj-10** [comment]
Hey @jacargentina, Thanks for the PR but this is adding lots of loops in order to achieve this. We may need to rethink this in a better way.  
Solution 1:- we may remove one option, keep only the label or value, and let the user handle it.
Solution 2:- Take `selectedLabel` props as well when setting up user control value and whenever the user clicks on Item we set that SelectedItem label in context to change the input value. 

Open to suggestion @jacargentina.

**Comment by jacargentina** [comment]
I've tried, but very confused about `useControllableState` on Select, and also the two context (SelectContext and SelectPortalContext)

**Comment by jacargentina** [comment]
@Viraj-10 there it is. Cheers!

File: packages/select/src/SelectInput.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const SelectInput = (StyledSelectInput: any) =>
         focusable={false}
         importantForAccessibility="no"
         placeholder={placeholder}
-        value={value ? value : ''}
+        value={label ? label : ''}
```
Comment (Viraj-10) [inline packages/select/src/SelectInput.tsx:19]: @jacargentina, Can we add fallback to Value if label is empty?

**Review by Viraj-10 — APPROVED** [review]

File: packages/select/src/SelectInput.tsx (hunk 1 above)
Comment (Viraj-10) [reply packages/select/src/SelectInput.tsx:19]: Also take pull from main

Comment (jacargentina) [reply packages/select/src/SelectInput.tsx]: Done!

**Comment by makkarMeenu** [comment]
Hey @jacargentina , thankyou for the PR. 
These changes work well for ios and android but this doesn't work for web. We haven't set label for the web in this PR. 
On web, handler should be on your select element. We will need to add the following code for web in `<select>` in 
`SelectPortal.tsx`

`setLabel(e.target.options[e.target.selectedIndex].text);`  

Please go ahead with the above change and I will merge this. Let me know in case of doubts.

**Comment by jacargentina** [comment]
Added!

**Comment by makkarMeenu** [comment]
> Added!

Hey, thanks!

**Pull request merged**